code/modules/power/singularity/emitter.dm

```diff
@@ -308,3 +335,128 @@
 		emagged = 1
 		if(user)
 			user.visible_message("[user.name] emags the [src.name].","<span class='notice'>You short out the lock.</span>")
+
+
+/obj/machinery/power/emitter/prototype
+	name = "Prototype Emitter"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "protoemitter"
+	can_buckle = TRUE
+	buckle_lying = 0
+	var/view_range = 12
+	var/datum/riding/riding_datum = null
+
+//BUCKLE HOOKS
+
+/obj/machinery/power/emitter/prototype/unbuckle_mob(mob/living/buckled_mob,force = 0)
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, 1)
+	for(var/obj/item/I in buckled_mob.held_items)
+		if(istype(I, /obj/item/weapon/turret_control))
+			qdel(I)
+	if(istype(buckled_mob))
+		buckled_mob.pixel_x = 0
+		buckled_mob.pixel_y = 0
+		if(buckled_mob.client)
+			buckled_mob.client.change_view(world.view)
+	for(var/datum/action/innate/protoemitter/H in buckled_mob.actions)
+		qdel(H)
+	. = ..()
+
+/obj/machinery/power/emitter/prototype/user_buckle_mob(mob/living/M, mob/living/carbon/user)
+	if(user.incapacitated() || !istype(user))
+		return
+	for(var/atom/movable/A in get_turf(src))
+		if(A.density)
+			if(A != src && A != M)
+				return
+	M.loc = get_turf(src)
+	..()
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, 1)
+	if(user.client)
+		user.client.change_view(view_range)
+	var/datum/action/innate/protoemitter/firing/auto = new()
+	auto.Grant(M, src)
+
+/datum/action/innate/protoemitter
+	check_flags = AB_CHECK_RESTRAINED | AB_CHECK_STUNNED | AB_CHECK_CONSCIOUS
+	var/obj/machinery/power/emitter/prototype/PE
+	var/mob/living/carbon/U
+
+
+/datum/action/innate/protoemitter/Grant(mob/living/L, obj/machinery/power/emitter/prototype/proto)
+	PE = proto
+	U = L
+	..()
+
+/datum/action/innate/protoemitter/firing
+	name = "Switch to Manual Firing"
+	desc = "The emitter will only fire on your command and at your designated target"
+	button_icon_state = "mech_zoom_off"
+
+/datum/action/innate/protoemitter/firing/Activate()
+	if(PE.manual == TRUE)
+		playsound(PE,'sound/mecha/mechmove01.ogg', 50, 1)
+		PE.manual = FALSE
+		name = "Switch to Manual Firing"
+		desc = "The emitter will only fire on your command and at your designated target"
+		button_icon_state = "mech_zoom_off"
+		for(var/obj/item/I in U.held_items)
+			if(istype(I, /obj/item/weapon/turret_control))
+				qdel(I)
+		UpdateButtonIcon()
+		return
+	if(PE.manual == FALSE)
+		playsound(PE,'sound/mecha/mechmove01.ogg', 50, 1)
+		name = "Switch to Automatic Firing"
+		desc = "Emitters will switch to periodic firing at your last target"
+		button_icon_state = "mech_zoom_on"
+		PE.manual = TRUE
+		for(var/V in U.held_items)

```

Comment:
Ff you want to drop everything he is helding, you can use `U.drop_all_held_items()` and then try to put in U's hands(with `put_in_hands(TC, TRUE)`, and only change the button's variables if the put in hands was true.